### ISSUE TITLE
[JENKINS-59865] Bogus Scala Compiler warning about missing files after test failure in sbt

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/analysis-model#1352]
<p>wsbt test run fails in Jenkins build, producing a line such as:</p>

```
[error] Total time: 127 s, completed Oct 7, 2019 9:11:30 AM

```
<p>Later, during</p>

```
post { always {  recordIssues enabledForFailure: true, tools: [scala()] } }

```
<p>the following output occurs in Jenkins log:</p>

```
[Scala Compiler] [-ERROR-] Can't resolve absolute paths for some files:
[Scala Compiler] [-ERROR-] - Total time: 127 s, completed Oct 7, 2019 9
[Scala Compiler] [-ERROR-] Can't create fingerprints for some files:
[Scala Compiler] [-ERROR-] - 'Total time: 127 s, completed Oct 7, 2019 9', IO exception has been thrown: java.nio.file.NoSuchFileException: Total time: 127 s, completed Oct 7, 2019 9

```

<p>Looks like a false positive from <span class="error">&#91;Sbt&#93;</span>ScalacParser regex capture at first glance:</p>

<p><a href="https://github.com/jenkinsci/analysis-model/blob/ddc3068166835508c4a5b1a8d30cec7d92109bfc/src/main/java/edu/hm/hafner/analysis/parser/SbtScalacParser.java#L19" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model/blob/ddc3068166835508c4a5b1a8d30cec7d92109bfc/src/main/java/edu/hm/hafner/analysis/parser/SbtScalacParser.java#L19</a></p>

<p><a href="https://github.com/jenkinsci/analysis-model/blob/ddc3068166835508c4a5b1a8d30cec7d92109bfc/src/main/java/edu/hm/hafner/analysis/parser/ScalacParser.java#L20" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model/blob/ddc3068166835508c4a5b1a8d30cec7d92109bfc/src/main/java/edu/hm/hafner/analysis/parser/ScalacParser.java#L20</a></p>

<p>Not a big deal at all, but can be confusing (in particular since scalac ran green).</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=proemer">proemer</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-59865" target="_blank">Bogus Scala Compiler warning about missing files after test failure in sbt</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jarredparrett">jarredparrett</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Trivial
<li><b>component(s)</b>: analysis-model
<li><b>label(s)</b>: help-wanted, newbie-friendly
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2020-01-07T10:47:30+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 20260108-221235
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>wsbt test run fails in Jenkins build, producing a line such as:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>[error] Total time: 127 s, completed Oct 7, 2019 9:11:30 AM
</pre>
</div></div>
<p>Later, during</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>post { always {  recordIssues enabledForFailure: true, tools: [scala()] } }
</pre>
</div></div>
<p>the following output occurs in Jenkins log:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>[Scala Compiler] [-ERROR-] Can't resolve absolute paths for some files:
[Scala Compiler] [-ERROR-] - Total time: 127 s, completed Oct 7, 2019 9
[Scala Compiler] [-ERROR-] Can't create fingerprints for some files:
[Scala Compiler] [-ERROR-] - 'Total time: 127 s, completed Oct 7, 2019 9', IO exception has been thrown: java.nio.file.NoSuchFileException: Total time: 127 s, completed Oct 7, 2019 9
</pre>
</div></div>

<p>Looks like a false positive from <span class="error">&#91;Sbt&#93;</span>ScalacParser regex capture at first glance:</p>

<p><a href="https://github.com/jenkinsci/analysis-model/blob/ddc3068166835508c4a5b1a8d30cec7d92109bfc/src/main/java/edu/hm/hafner/analysis/parser/SbtScalacParser.java#L19" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model/blob/ddc3068166835508c4a5b1a8d30cec7d92109bfc/src/main/java/edu/hm/hafner/analysis/parser/SbtScalacParser.java#L19</a></p>

<p><a href="https://github.com/jenkinsci/analysis-model/blob/ddc3068166835508c4a5b1a8d30cec7d92109bfc/src/main/java/edu/hm/hafner/analysis/parser/ScalacParser.java#L20" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model/blob/ddc3068166835508c4a5b1a8d30cec7d92109bfc/src/main/java/edu/hm/hafner/analysis/parser/ScalacParser.java#L20</a></p>

<p>Not a big deal at all, but can be confusing (in particular since scalac ran green).</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins v2.190.1<br/>
Warnings-NG v6.1.1
```
</details>
<details><summary><i>1 attachment</i></summary>

- [Bug_1_Diagram.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/49845/Bug_1_Diagram.jpg)
  > ![Bug_1_Diagram.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/49845/Bug_1_Diagram.jpg)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-59865?focusedId=382329&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-382329">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I think it is much simpler to reproduce and fix this bug. You only need to change and test the analysis model module. Create a text file that contains the message from above (with the error) and read it in the test class <tt>SbcParserTest</tt> with the parser <tt>SbcParser</tt>. It should produce a warning. Then fix the bug so that no warning will be shown anymore.</p>

<p>There is no need to use the warnings-ng plugin for the test, as the bug is only part of the library analysis-model.</p>
</pre>
</details>

</ul>
</details>

<p>I think it is much simpler to reproduce and fix this bug. You only need to change and test the analysis model module. Create a text file that contains the message from above (with the error) and read it in the test class <tt>SbcParserTest</tt> with the parser <tt>SbcParser</tt>. It should produce a warning. Then fix the bug so that no warning will be shown anymore.</p>

<p>There is no need to use the warnings-ng plugin for the test, as the bug is only part of the library analysis-model.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jarredparrett:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-59865?focusedId=382342&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-382342">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thank you for the direction - it made it much easier to recreate. In working on it, I think its test class was SbtScalacParserTest.java; I ran that with a custom text file that contained the content from sbtScalac.txt with the addition of the above line. The test then fails as it expects 4 errors in the report but gets 5. To resolve this, I changed the SbtScalacParser.java to include the negative lookahead statement - </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-quote">"^(\\[warn\\]|\\[error\\](?!\\s+Total\\stime:))\\s*(.*?):(\\d+)(?::\\d+)?:\\s*(.*)$"</span>
</pre>
</div></div>


<p> Thank you for your help</p>
</pre>
</details>

</ul>
</details>

<p>Thank you for the direction - it made it much easier to recreate. In working on it, I think its test class was SbtScalacParserTest.java; I ran that with a custom text file that contained the content from sbtScalac.txt with the addition of the above line. The test then fails as it expects 4 errors in the report but gets 5. To resolve this, I changed the SbtScalacParser.java to include the negative lookahead statement - </p>

<pre>
<span class="code-quote">"^(\\[warn\\]|\\[error\\](?!\\s+Total\\stime:))\\s*(.*?):(\\d+)(?::\\d+)?:\\s*(.*)$"</span>
</pre>


<p> Thank you for your help</p>
